### PR TITLE
Allow expressions for text-color in format expression

### PR DESCRIPTION
**Pull request by persidskiy**
Adds ability to use expressions for `text-color` property in `format` expression. https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions/#types-format

## Pull request checklist:
 - [x] Describe the changes in this PR, especially public API changes.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. Put tests in correct [Test Plan](https://github.com/mapbox/mapbox-maps-ios/tree/main/Tests/TestPlans) (Unit, Integration, All)
   - [ ] If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.
 - [ ] Add any new public, top-level symbols to the Jazzy config's `custom_categories` (scripts/doc-generation/.jazzy.yaml)
 - [ ] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] Update the guides (internal access only), README.md, and DEVELOPING.md if their contents are impacted by these changes.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` first and then port to `v10.[version]` release branch.

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

[MAPSIOS-520](https://mapbox.atlassian.net/browse/MAPSIOS-520)

### CODE REVIEW

File: Sources/MapboxMaps/Style/Types/ExpressionOptions.swift [hunk 1]
```diff
@@ -64,14 +64,21 @@ public struct FormatOptions: Codable, Equatable, ExpressionArgumentConvertible {
...
 
-    public init(fontScale: Double? = nil, textFont: [String]? = nil, textColor: UIColor? = nil) {
+    public init(fontScale: Double? = nil, textFont: [String]? = nil, textColor: Value<StyleColor>? = nil) {
         self.fontScale = fontScale
         self.textFont = textFont
 
         if let textColor = textColor {
-            self.textColor = StyleColor(textColor)
+            self.textColor = textColor
         }
```
Comment (maios) [inline Sources/MapboxMaps/Style/Types/ExpressionOptions.swift:73]: why not just assign textColor here? we don't need textColor to be non-nil, this check belongs to the other initializer where the parameter for `textColor` is a `UIColor`

File: Sources/MapboxMaps/Style/Types/ExpressionOptions.swift [hunk 2]
```diff
@@ -52,7 +52,7 @@ public struct FormatOptions: Codable, Equatable, ExpressionArgumentConvertible {
     public var textFont: [String]?
 
     /// Overrides the color specified by the root paint property.
-    public var textColor: StyleColor?
+    public var textColor: Value<StyleColor>?
```
Comment (maios) [inline Sources/MapboxMaps/Style/Types/ExpressionOptions.swift:55]: this is breaking change I supposed, I wonder if this is something we can accept @evil159 @OdNairy @pjleonard37 ?

Comment (evil159) [reply Sources/MapboxMaps/Style/Types/ExpressionOptions.swift:55]: v11?

File: Sources/MapboxMaps/Style/Types/Value.swift [hunk 3]
```diff
@@ -34,6 +34,19 @@ public enum Value<T: Codable>: Codable {
...
 
-extension Value: Equatable where T: Equatable {
+extension Value: Equatable where T: Equatable {}
 
+extension Value {
+    internal init(constant: T) {
+        self = .constant(constant)
+    }
+
+    var asConstant: T? {
```
Comment (maios) [inline Sources/MapboxMaps/Style/Types/Value.swift:44]: nit
```suggestion
    internal var asConstant: T? {
```

**Review by maios — APPROVED** [review]

**Pull request merged**